### PR TITLE
lower default SimpleTimeCluster straw/panel count thresholds

### PR DESCRIPTION
**Pull request by edcallaghan**
This PR relaxes the occupancy thresholds to construct `TimeCluster`s in `SimpleTimeCluster`, which seeds straight-line cosmic reconstruction. Using "bare minimum" thresholds as implemented here (>= 1 panel, >= 2 straws), improves the efficiency for near-verticals from ~1% to ~10%. This is shown in the below plot, which can be cross-referenced against cosmic reco nightly validation.
cosmics-all-tracks-counts-vs-costheta-log

### CODE REVIEW

File: CosmicReco/src/SimpleTimeCluster_module.cc [hunk 1]
```diff
@@ -41,7 +41,7 @@ class SimpleTimeCluster : public art::EDProducer {
     using Name = fhicl::Name;
     using Comment = fhicl::Comment;
     fhicl::Atom<int> debug{Name("debugLevel"), Comment("set to 1 for debug prints"), 0};
-    fhicl::Atom<int> minnsh{Name("minNStrawHits"), Comment("minimum number of straw hits "), 5};
+    fhicl::Atom<int> minnsh{Name("minNStrawHits"), Comment("minimum number of straw hits "), 2};
```
Comment (brownd1978) [inline CosmicReco/src/SimpleTimeCluster_module.cc:44]: code defaults of algorithm parameters is deprecated.  Please remove this default and add the parameter to the prolog.  While you're in there, please do the same fo the other parameters.

**Review by brownd1978 — CHANGES_REQUESTED** [review]

**Comment by edcallaghan** [comment]
@FNALbuild run build test

**Comment by edcallaghan** [comment]
@FNALbuild run build test

**Comment by brownd1978** [comment]
There are errors in the new prolog causing the CI to fail.  In case you didn't know, you can't see the CI logs in a browser, but you can copy the links to them from GitHub and view them using 'curl' once logged on to a mu2egpvm* machine.

Failed to parse the configuration file 'Production/Validation/ceSimReco.fcl' with exception
---- Parse error BEGIN
  detected at or near line 44, character 19, of file "/scratch/workspace/GitHubPRTests/mu2e-offline-build-test/Offline/CosmicReco/fcl/prolog.fcl"
  included from line 15 of file "/scratch/workspace/GitHubPRTests/mu2e-offline-build-test/Production/JobConfig/reco/prolog.fcl"
  included from line 11 of file "./Production/Validation/ceSimReco.fcl"
  
      debugLevel : 0,
                    ^
---- Parse error END

**Comment by edcallaghan** [comment]
@brownd1978 The prolog errors were the first failure, which I pushed a fix for. The latest jobs all failed non-gracefully with an underlying system error. For the art jobs, this presents as `*** Error in `mu2e': double free or corruption (!prev): 0x0000000005283d80 ***`, after the modules all run to completion.

It looks like there is an alternate `SimpleTimeCluster` defined in `mu2e_trig_config`, which will need to either inherit from here or have its configuration updated to accommodate the default values before some of the validaiton jobs here will succeed. But I don't think that that's behind the current slew of errors.

**Comment by edcallaghan** [comment]
And, FWIW, after fixing that alternate `SimpleTimeCluster` configuration, the jobs (e.g. `ceSteps` and `ceDigi`) run fine locally. I'm going to trigger the tests again just to see what happens...

**Comment by edcallaghan** [comment]
@FNALbuild run build test

**Comment by edcallaghan** [comment]
Ok, not sure what went wrong with the last batch. But the three failing jobs are dependent on `mu2e_trig_config`. I'll submit a PR there to update the upstream configuration.

**Comment by kutschke** [comment]
Ed has PR #28 in to mu2e_trig_config.  I would like him to change something in it but it should fix the problems as is.

I forget if we updated the CI to use mu2e_trig_config in test with. Let's see.

@FNALbuild run build test with Mu2e/mu2e_trig_config#28

**Comment by kutschke** [comment]
You may already know this but just to be sure.   To test interactively, make sure you have all of Offline, Production and mu2e_trig_config in your muse working area ( if not, log out, clone the missing repos, login, mu2einit, muse setup and build ).  Then you can test the first one interactively with :

`
 mu2e -c Production/Validation/cosmicOffSpill.fcl -n 10`

The next one is a bit trickier since it uses the output of ceSteps and looks for it locally

```
 mu2e -c Production/Validation/ceSteps.fcl -n 10
 mu2e -c Production/Validation/ceDigi.fcl -n 10
```

The same for testing ceMix, it consumes the output of muDauSteps, which does not write output for event event.  So need to run enough to be sure.

'''
 mu2e -c Production/Validation/muDauSteps.fcl -n 100
  mu2e -c Production/Validation/ceMix.fcl -n 1
'''

Be sure to clean up the .art files in your working area when you are done.

If you get errors with text "root://fndcadoor.fnal. .... " you need a certiifcate to access files via xrootd.  Use "vomsCert".  See https://mu2ewiki.fnal.gov/wiki/Authentication#Certificate  and https://mu2ewiki.fnal.gov/wiki/DataTransfer#xrootd .

**Comment by kutschke** [comment]
Hmmm.  Test with was not helpful.

**Comment by edcallaghan** [comment]
@kutschke do you have a theory as to what's causing these slates of memory-related errors? Are these jobs running on AL9? The memory-map printout references an SL7 build, and when I test locally on an lbl machine running SL7 the jobs are successful.

**Comment by kutschke** [comment]
@edcallaghan  My experiment failed and we can ignore it.  

I suggest making the changes I suggested in mu2e_trig_config and testing locally.  If that works push to your mu2e_trig_config fork and we can merge mu2e_trig_config and rerun the CI.

**Comment by edcallaghan** [comment]
Ok, well, with the latest commit to `mu2e_trig_config` which properly extends the `CstSimpleTimeCluster` defined therein from the default defined here in Offline, all of relevant validation jobs (i.e. `ceSteps`, `muDauSteps`, `ceDigi`, `ceMix`, and `cosmicOffSpill`) run and terminate without any errors.

**Comment by kutschke** [comment]
Created issue #1277 to followup on adding an mock namespace to  CosmicReco/fcl/prolog.fcl.  That should be a separate PR.

**Review by kutschke — APPROVED** [review]

**Comment by kutschke** [comment]
@edcallaghan since yesterday afternoon there have been intermittent issues with access to the cached DB server.  So any CI you run now will likely fail.   Our support is 5/8 but sometimes we get lucky over weekends.

**Comment by kutschke** [comment]
The PR in mu2e_trig_config has been merged.  So the CI should work now

@FNALbuild run build test

**Pull request merged**